File: mobile_ui/quickchat_ui/lib/features/chat/data/dto/GetConversations.dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'GetConversations.dto.freezed.dart';

@freezed
class GetConversationsDto with _$GetConversationsDto {
  const factory GetConversationsDto({
    required String conversationsId,
    required int page,
  }) = _GetConversationsDto;

  const GetConversationsDto._();

  Map<String, dynamic> toPersistence() {
    return <String, dynamic>{
      'conversationsId': conversationsId,
      'page': page,
    };
  }
}

File: backend/lara_be/app/Domain/Chat/DTO/GetConversationsDto.php
<?php

namespace App\Domain\Chat\DTO;

class GetConversationsDto
{
    public string $conversationsId;
    public int $page;

    public function __construct(string $conversationsId, int $page)
    {
        $this->conversationsId = $conversationsId;
        $this->page = $page;
    }
}
